pandas/tseries/index.py

```diff
@@ -1257,6 +1257,12 @@ def get_loc(self, key, method=None):
                                           'when key is a time object')
             return self.indexer_at_time(key)
 
+        from pandas.tseries.period import Period
+
+        if isinstance(key, Period):
```

Comment:
use ``isinstance(key, ABCPeriod)`` (and import this at the top). This avoids the additional import here.